core/domain/user_services.py

```diff
@@ -44,7 +49,10 @@ def __init__(
         self.last_agreed_to_terms = last_agreed_to_terms
         self.last_started_state_editor_tutorial = (  # pylint: disable=invalid-name
             last_started_state_editor_tutorial)
-        self.profile_picture_data_url = profile_picture_data_url
+        if profile_picture_data_url:
+            self.profile_picture_data_url = profile_picture_data_url
+        else:
+            self.profile_picture_data_url = self.fetch_gravatar(email)
```

Comment:
No -- gravatar fetching should only happen once, ever, in the lifecycle of a user. We instantiate user objects very frequently and I don't think there should be overhead here. (In addition, __init__ isn't really the right place to do this sort of thing anyway.)